### PR TITLE
Apply private to symbols in IO module marked with pragma "no doc"

### PR DESCRIPTION
**Pull request by lydia-duncan**
Once chpldoc recognizes private symbols, the pragmas can be removed.

Passed full testing

### CODE REVIEW

File: modules/standard/IO.chpl [hunk 1]
```diff
@@ -1368,33 +1368,33 @@ pragma "no doc"
...
+private extern const QIO_HINT_CACHED:c_int;
 pragma "no doc"
-extern const QIO_HINT_PARALLEL:c_int;
+private extern const QIO_HINT_PARALLEL:c_int;
 pragma "no doc"
-extern const QIO_HINT_DIRECT:c_int;
+private extern const QIO_HINT_DIRECT:c_int;
 pragma "no doc"
-extern const QIO_HINT_NOREUSE:c_int;
+private extern const QIO_HINT_NOREUSE:c_int;
```
Comment (mppf) [inline modules/standard/IO.chpl]: I think these should remain public (and uncodumented) for now. They're all features you can use, they just don't have reasonable names. (QIO_FDFLAG_ and QIO_METHOD_)

File: modules/standard/IO.chpl [hunk 2]
```diff
@@ -1588,255 +1588,255 @@ pragma "no doc"
...
+private extern proc qio_channel_scan_literal(threadsafe:c_int, ch:qio_channel_ptr_t, const match:c_string, len:ssize_t, skipwsbefore:c_int):syserr;
 pragma "no doc"
-extern proc qio_channel_scan_literal_2(threadsafe:c_int, ch:qio_channel_ptr_t, match:c_void_ptr, len:ssize_t, skipwsbefore:c_int):syserr;
+private extern proc qio_channel_scan_literal_2(threadsafe:c_int, ch:qio_channel_ptr_t, match:c_void_ptr, len:ssize_t, skipwsbefore:c_int):syserr;
 pragma "no doc"
-extern proc qio_channel_print_literal(threadsafe:c_int, ch:qio_channel_ptr_t, const match:c_string, len:ssize_t):syserr;
+private extern proc qio_channel_print_literal(threadsafe:c_int, ch:qio_channel_ptr_t, const match:c_string, len:ssize_t):syserr;
 pragma "no doc"
-extern proc qio_channel_print_literal_2(threadsafe:c_int, ch:qio_channel_ptr_t, match:c_void_ptr, len:ssize_t):syserr;
+private extern proc qio_channel_print_literal_2(threadsafe:c_int, ch:qio_channel_ptr_t, match:c_void_ptr, len:ssize_t):syserr;
```
Comment (mppf) [inline modules/standard/IO.chpl]: These look good.

File: modules/standard/IO.chpl [hunk 3]
```diff
@@ -1874,82 +1874,82 @@ extern record qio_conv_t {
...
-extern proc qio_format_error_too_few_args():syserr;
+private extern proc qio_format_error_too_few_args():syserr;
 pragma "no doc"
-extern proc qio_format_error_arg_mismatch(arg:int):syserr;
+private extern proc qio_format_error_arg_mismatch(arg:int):syserr;
 pragma "no doc"
 extern proc qio_format_error_bad_regexp():syserr;
 pragma "no doc"
-extern proc qio_format_error_write_regexp():syserr;
+private extern proc qio_format_error_write_regexp():syserr;
```
Comment (mppf) [inline modules/standard/IO.chpl]: Looks good here too.

File: modules/standard/IO.chpl [hunk 4]
```diff
@@ -6862,13 +6862,13 @@ iter channel.matches(re:regexp, param captures=0, maxmatches:int = max(int))
...
+private extern const FTYPE_NONE   : c_int;
 pragma "no doc"
-extern const FTYPE_HDFS   : c_int;
+private extern const FTYPE_HDFS   : c_int;
 pragma "no doc"
-extern const FTYPE_LUSTRE : c_int;
+private extern const FTYPE_LUSTRE : c_int;
 pragma "no doc"
-extern const FTYPE_CURL   : c_int;
+private extern const FTYPE_CURL   : c_int;
```
Comment (mppf) [inline modules/standard/IO.chpl]: Good!

**Comment by mppf** [comment]
Mostly, looks good, please leave the QIO_FDFLAG_ and QIO_METHOD_ constants public for now.

File: modules/standard/IO.chpl (hunk 1 above)
Comment (lydia-duncan) [inline modules/standard/IO.chpl]: I'm assuming that includes the QIO_HINTs?

Comment (mppf) [inline modules/standard/IO.chpl]: yes

**Pull request merged**